javalib/src/main/scala/java/net/URI.scala

```diff
@@ -41,18 +41,19 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     else fld(AbsHierPart)
   }.get
 
-  private val _authority = fld(AbsAuthority, RelAuthority)
+  private val _authority = fld(AbsAuthority, RelAuthority).filter(_ != "")
   private val _userInfo = fld(AbsUserInfo, RelUserInfo)
   private val _host = fld(AbsHost, RelHost)
   private val _port = fld(AbsPort, RelPort).fold(-1)(_.toInt)
 
   private val _path = {
-    if (_isAbsolute) {
-      if (_authority.isDefined) fld(AbsNetPath)
-      else fld(AbsAbsPath)
+    if (fld(AbsAuthority, RelAuthority).isDefined) {
+      val netPath = fld(if (_isAbsolute) AbsNetPath else RelNetPath)
+      netPath orElse ("": js.UndefOr[String])
```

Comment:
replace by: `fld(AbsNetPath, RelNetPath)`
`net_path` contains the authority. So if the authority is non-empty, so is `net_path`.